### PR TITLE
Chore: Bump Storybook to 8.4.x

### PR DESCRIPTION
**Pull request by jackw**
**What is this feature?**

Updates storybook to latest (8.4.x) and removes the patch for storybook/blocks as its become so complex to bump storybook it seems better to just remove it and switch out all the mdx `preview` usage for ~`canvas`~ a `ExampleFrame` component.

**Why do we need this feature?**

To keep us up to date and make it far easier to update in the future as we remove the tech debt related to patching the blocks package. Refer to the [storybook issue](https://github.com/storybookjs/storybook/issues/29576) to understand the need for `copyAssets` added in this PR.

**Who is this feature for?**

Grafana contributors and plugin developers

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Comment by joshhunt** [comment]
Pushed up changes:
 - Removes `@storybook/addon-mdx-gfm`. This is a compat package that should be [migrated away from](https://storybook.js.org/docs/writing-docs/mdx#markdown-tables-arent-rendering-correctly), but it looks like we don't actually use Github Flavoured Markdown, so I just removed it completely. I did a brief check over the docs and it looked fine, but I may have missed some things.
 - Added an `<ExampleFrame />` component to replace instances of using `<Preview />` to render custom components. ExampleFrame is pretty simple, it just renders children with a border, but it also makes store Storybook styles don't interfere with the components
 - Modified `ThemedDocsContainer` to include our GlobalStyles to make sure some components render correctly.
 - Went through and improved docs slightly to use the above changes

File: packages/grafana-ui/.storybook/main.ts [hunk 1]
```diff
@@ -45,17 +34,18 @@ const mainConfig: StorybookConfig = {
...
+          sassOptions: {
+            // silencing these warnings since we're planning to remove sass when angular is gone
+            silenceDeprecations: ['import', 'global-builtin'],
+          },
+        },
       },
     },
     getAbsolutePath('@storybook/addon-storysource'),
     getAbsolutePath('storybook-dark-mode'),
-    getAbsolutePath('@storybook/addon-mdx-gfm'),
```
Comment (tskarhed) [inline packages/grafana-ui/.storybook/main.ts:52]: Is gfm supported by default now?

File: packages/grafana-ui/src/components/Combobox/MultiCombobox.internal.story.tsx [hunk 2]
```diff
@@ -1,16 +1,10 @@
...
-import mdx from './MultiCombobox.mdx';
 
 const meta: Meta<typeof MultiCombobox> = {
   title: 'Forms/MultiCombobox',
   component: MultiCombobox,
-  parameters: {
-    docs: {
-      page: mdx,
-    },
-  },
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/Combobox/MultiCombobox.internal.story.tsx:13]: Is this not needed anymore? The doc page seems to look okay.

File: packages/grafana-ui/src/components/Link/TextLink.story.tsx [hunk 3]
```diff
@@ -32,7 +32,7 @@ const meta: Meta = {
     inline: { control: 'boolean' },
   },
   args: {
-    href: 'https://www.google.com',
+    href: 'https://www.grafana.com',
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/Link/TextLink.story.tsx:35]: 😄

File: packages/grafana-ui/src/utils/storybook/ExampleFrame.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+interface ExampleFrameProps {
+  children: ReactNode;
+}
+
+/**
+ * Wraps children with a border for nicer presentation in Storybook docs.
+ *
+ * This is intended to be used temporarily to move away from our <Preview> patch until
+ * we have something more long term.
```
Comment (tskarhed) [inline packages/grafana-ui/src/utils/storybook/ExampleFrame.tsx:17]: Longer term, is this a general strategy around how we do docs, including Examples or not?

File: .betterer.results [hunk 5]
```diff
@@ -5397,6 +5397,9 @@ exports[`no undocumented stories`] = {
     "packages/grafana-ui/src/components/ThemeDemos/ThemeDemo.story.tsx:5381": [
       [0, 0, 0, "No undocumented stories are allowed, please add an .mdx file with some documentation", "5381"]
     ],
+    "packages/grafana-ui/src/components/UnitPicker/UnitPicker.story.tsx:5381": [
+      [0, 0, 0, "No undocumented stories are allowed, please add an .mdx file with some documentation", "5381"]
+    ],
```
Comment (tskarhed) [inline .betterer.results:5402]: Do we want an mdx file here? If not, I think we should rethink this betterer rule in general

File: packages/grafana-ui/src/utils/storybook/ExampleFrame.tsx (hunk 4 above)
Comment (joshhunt) [reply packages/grafana-ui/src/utils/storybook/ExampleFrame.tsx:17]: I'm hoping the longer term strategy would be to make stories for everything we want examples in docs for.

File: .betterer.results (hunk 5 above)
Comment (joshhunt) [reply .betterer.results:5402]: An empty MDX file serves no one. Ash and I have discussed removing the betterer rule, but I didn't want to do that in this PR.

File: packages/grafana-ui/src/components/Link/TextLink.story.tsx (hunk 3 above)
Comment (joshhunt) [reply packages/grafana-ui/src/components/Link/TextLink.story.tsx:35]: 🏆

File: packages/grafana-ui/.storybook/main.ts (hunk 1 above)
Comment (joshhunt) [reply packages/grafana-ui/.storybook/main.ts:52]: Nope. This package is supposed to be a compat package, and we should move to another one, but we don't actually use GFM so I removed it all together.

File: .betterer.results (hunk 5 above)
Comment (jackw) [reply .betterer.results:5402]: This one wasn't empty, it was broken with runtime error messages in storybook. The MDX file only had argtypes which the auto generated docs have so seemed simpler to just delete it for now. 🤷

**Review by jackw — COMMENTED** [review]
I'm approving my own PR (as much as I can with this comment)! 🚀 

Thanks so much for the support with this @joshhunt it looks great. 🥳

**Review by joshhunt — APPROVED** [review]
I'm also approving this despite writing the other half of it. I'll check with someone else just to be sure :)

**Review by tskarhed — APPROVED** [review]

**Pull request merged**